### PR TITLE
fix: spin build failed

### LINKED ISSUE #108: An error occurred when executing make in version 0.7.0
I tried to use the make command to build the program and encountered an error：

image

image

### PR DESCRIPTION
**Pull request by dierbei**
Fix #108

### CODE REVIEW

**Comment by Mossaka** [comment]
Thanks for your contribution! 

This does resolve the problem that #108 brings. However, I'd love to propose an alternative way - instead of adding another target at the `build-spin` step, could we add it to a `install-rust-targets` target that does something similar to https://github.com/deislabs/containerd-wasm-shims/blob/main/.github/workflows/build.yaml#L30-L33? 

This allows us to reuse the same step in both the CI and local dev environment.

File: Makefile [hunk 1]
```diff
@@ -55,7 +55,7 @@ build-wws-cross-%: install-cross
 
 .PHONY: build-spin
 build-spin:
-	cargo build --release --manifest-path=containerd-shim-spin-v1/Cargo.toml
+	rustup target add wasm32-unknown-unknown && cargo build --release --manifest-path=containerd-shim-spin-v1/Cargo.toml
```
Comment (Mossaka) [inline Makefile:58]: I propose to add new step called `install-rust-targets` that adds both wasm32-unknown-unknown and wasm32-wasi targets, and make it a prerequisite for `build-spin` and `build-slight`.

**Comment by dierbei** [comment]
@microsoft-github-policy-service agree

File: Makefile (hunk 1 above)
Comment (dierbei) [reply Makefile]: Thanks for your advice. I modified a few places:
1. Makefile added install-rust-targets
2. build.yaml and ci.yaml use install-rust-targets

**Comment by Mossaka** [comment]
@dierbei could you please sign your commits?

**Comment by dierbei** [comment]
@Mossaka I tried doing a rebase so maybe the commit will be clearer.

**Pull request merged**